src/tcgwars/logic/impl/gen7/SunMoonPromos.groovy

```diff
@@ -2619,9 +2677,102 @@ public enum SunMoonPromos implements LogicCardInfo {
         return copy(UnbrokenBonds.LUCARIO_MELMETAL_GX_120, this);
       case GARCHOMP_GIRATINA_GX_SM193:
         return copy(UnifiedMinds.GARCHOMP_GIRATINA_GX_146, this);
+      case DETECTIVE_PIKACHU_SM194:
+        return basic (this, hp:HP090, type:L, retreatCost:2) {
+        weakness F
+        resistance M, MINUS20
+        move "Brilliant Deduction", {
+          text "Look at the top 4 cards of your deck and put 1 of them into your hand. Shuffle the other cards back into your deck."
+          energyCost L
+          attackRequirement {
+            assert my.deck : "Your deck is empty."
+          }
+          onAttack {
+            def top4 = my.deck.subList(0,4)
+            top4.showToMe("The top 4 cards of your deck.")
+            top4.select(max:1,"Choose a card to put in your hand.").moveTo(my.hand)
+            shuffleDeck()
+            }
+          }
+        };
+      case CHARIZARD_GX_SM195:
+        return evolution (this, from:"Charmeleon", hp:HP250, type:R, retreatCost:4) {
+          weakness W
+          move "Raging Destruction", {
+            text "Discard the top 8 cards of your deck. If any of those cards are [R] Energy cards, attach them to this Pokémon."
+            energyCost R
+            attackRequirement {
+              assert my.deck : "Your deck is empty."
+            }
+            onAttack {
+              def top8 = my.deck.subList(0,8)
+              def list = top8.filterByType(BASIC_ENERGY).filterByEnergyType(R)
+              def num = list.size()
+              top8.showToMe("The top 8 cards of your deck.")
+              list.showToMe("These [R] Energies will be attached.")
```

Comment:
This could be moved inside the `if (num)` block, we don't have to show an empty list if there's no [R]

Reply:
Good point. Made this adjustment